### DISCUSSION TITLE
Secrets for pipeline

### DISCUSSION THREAD

**Discussion by tuomassiren** [aws/copilot-cli#4344] [General]
How do I tell the pipeline build stage a token for private NPM repository?

**Comment by behdad088** [comment]
To use a token for a private NPM repository in AWS pipeline build stage, you can follow these steps:

Create an access token in your private NPM repository provider.
In your AWS CodePipeline, go to the build stage for your project.
In the build stage, click on "Edit stage".
In the "Edit build stage" dialog box, expand the "Environment" section.
In the "Environment" section, click on "Add environment variable".
In the "Add environment variable" dialog box, enter a name for your environment variable (e.g. "NPM_TOKEN") and paste in your access token for the private NPM repository.
Click on "Save".
In your build script, reference the environment variable using the syntax ${ENV_VARIABLE_NAME} (e.g. ${NPM_TOKEN}) to authenticate with the private NPM repository during the build process.
This will allow your AWS pipeline to access your private NPM repository using the access token you provided.
